packages/sdk-js/scripts/codegen.js

```diff
@@ -0,0 +1,370 @@
+
+const COMMENT_HEADER = "/* @generated by codegen. DO NOT EDIT BY HAND */";
+
+const VERSIONED_ACTIVITY_TYPES = {
+  "ACTIVITY_TYPE_CREATE_AUTHENTICATORS": "ACTIVITY_TYPE_CREATE_AUTHENTICATORS_V2",
+  "ACTIVITY_TYPE_CREATE_PRIVATE_KEYS": "ACTIVITY_TYPE_CREATE_PRIVATE_KEYS_V2",
+  "ACTIVITY_TYPE_CREATE_SUB_ORGANIZATION": "ACTIVITY_TYPE_CREATE_SUB_ORGANIZATION_V4",
+  "ACTIVITY_TYPE_CREATE_USERS": "ACTIVITY_TYPE_CREATE_USERS_V2",
+  "ACTIVITY_TYPE_SIGN_RAW_PAYLOAD": "ACTIVITY_TYPE_SIGN_RAW_PAYLOAD_V2",
+  "ACTIVITY_TYPE_SIGN_TRANSACTION": "ACTIVITY_TYPE_SIGN_TRANSACTION_V2",
+}
+
+// Helper Functions
+/**
+ * @param {Array<string | null>} input
+ * @returns {string}
+ */
+function joinPropertyList(input) {
+  return input.filter(Boolean).join(",\n");
+}
+
+/**
+ * @param {string} methodName
+ * @returns {string}
+ */
+function methodTypeFromMethodName(methodName) {
+  if (["approveActivity", "rejectActivity"].includes(methodName)) { return "activityDecision"; }
+  if (methodName.startsWith("nOOP")) { return "noop"; }
+  if (methodName.startsWith("get") || methodName.startsWith("list")) { return "query"; }
+  return "command";
+}
+
+// Generators
+const generateApiTypesFromSwagger = async (swaggerSpec, targetPath) => {
+  const namespace = swaggerSpec.tags?.find((item) => item.name != null)?.name;
+
+  /** @type {Array<string>} */
+  const codeBuffer = [];
+
+  /** @type {Array<string>} */
+  const imports = [];
+
+  imports.push(
+    'import type { operations } from "../__inputs__/public_api.types";'
+  )
+
+  for (const endpointPath in swaggerSpec.paths) {
+    const methodMap = swaggerSpec.paths[endpointPath];
+    const operation = methodMap.post;
+    const operationId = operation.operationId;
+
+    const operationNameWithoutNamespace = operationId.replace(
+      new RegExp(`${namespace}_`),
+      ""
+    );
+
+    const isEndpointDeprecated = Boolean(operation.deprecated);
+
+    const methodName = `${
+      operationNameWithoutNamespace.charAt(0).toLowerCase() +
+      operationNameWithoutNamespace.slice(1)
+    }`;
+
+    const methodType = methodTypeFromMethodName(methodName);
+    const signedRequestGeneratorName = `sign${methodName}`;
+    const parameterList = operation["parameters"] ?? [];
+
+    // let responseValue = "void";
+    // if (methodType === "command") {
+    //   responseValue = `operations["${operationId}"]["responses"]["200"]["schema"]["activity"]["result"]["${methodName}Result"]`;
+    // } else if (["noop", "query"].includes(methodType)) {
+    //   responseValue = `operations["${operationId}"]["responses"]["200"]["schema"]`;
+    // } else if (methodType === "activityDecision") {
+    //   responseValue = `operations["${operationId}"]["responses"]["200"]["schema"]["activity"]["result"]`;
+    // }
+
+    let responseValue = "void";
+    if (methodType === "command") {
+      responseValue = `operations["${operationId}"]["responses"]["200"]["schema"]`;
```

Comment:
these all seem to be equivalent. do we mean to keep the commented out version above ^?